packages/compiler-cli/src/transformers/compiler_host.ts

```
@@ -211,12 +217,28 @@ export class TsCompilerAotCompilerTypeCheckHostAdapter implements ts.CompilerHos
         importedFile = rootedImportedFile;
       }
       moduleName = dotRelative(path.dirname(containingFile), importedFile);
-    } else if (importedFilePackagName) {
+    } else if (importedFilePackageName) {
       moduleName = stripNodeModulesPrefix(importedFile);
+      if (originalImportedFile.endsWith('.d.ts')) {
+        // the moduleName for these typings could be shortented to the npm package name
+        // if the npm package typings matches the importedFile
+        try {
+          const modulePath = importedFile.substring(0, importedFile.length - moduleName.length) +
+              importedFilePackageName;
+          const packageJson = require(modulePath + '/package.json');
+          const packageTypings = path.posix.join(modulePath, packageJson.typings);
+          if (packageTypings === originalImportedFile) {
+            moduleName = importedFilePackageName;
+          }
+        } catch (e) {
```

Leave a comment explaining why ignoring the error is okay.

---

Done